### ISSUE TITLE
[BUG] useShow hook id 

### ISSUE THREAD

**Issue by Hori9600** [pankod/refine#1701]
Same bug as issue #1604, but now with the useShow hook

**Comment by omeraplak** [comment]
Hey @Hori9600 👋,
Thank you for contacting us. Can you share a CodeSandbox or sample code so we can reproduce the error? Thank you!

**Comment by Hori9600** [comment]
Hi,
I think there is a problem with the useShow hook.
I think this line:
```
const [showId, setShowId] = useState<BaseKey | undefined>(
        id ?? idFromRoute,
    );
```

should be:
   ```
 const defaultId =
        !resourceFromProp || resourceFromProp === routeResourceName
            ? id ?? idFromRoute
            : undefined;

    const [showId, setShowId] = useState<BaseKey | undefined>(defaultId);

```

**Comment by omeraplak** [comment]
Hey @Hori9600 ,
Thank you! I've created PR #1702 🚀

**Comment by omeraplak** [comment]
Hey @Hori9600 ,
We've released `3.12.1`. Could you approve this fix?

**Comment by Hori9600** [comment]
Yes, it was quick. Thank you
